### DISCUSSION TITLE
How does subfinder work, what are "passive online sources"?

### DISCUSSION THREAD

**Discussion by sb-89** [projectdiscovery/subfinder#1253] [Q&A]
> subfinder is a subdomain discovery tool that returns valid subdomains for websites, using passive online sources

What are passive online sources? Are subdomains found some programmatic way or how is the source of subdomains gotten?

Thanks

**Comment by GeorginaReeder** [comment]
Hi @sb-89 ! This related article on Subfinder might be helpful: https://blog.projectdiscovery.io/do-you-really-know-subfinder-an-in-depth-guide-to-all-features-of-subfinder-beginner-to-advanced/

It goes into a bit more detail about the passive sources we use, such as Binaryedge, C99, and others.

If you have any other questions though, just let us know.

**Comment by Bundy01** [comment]
Hi,

I use subfinder as follows
`subfinder -nW -r 9.9.9.10,1.1.1.1 -dL FILE -o OUT`
Is this enough to find all valid subdomains (I'm using it with the original configuration)?

What does the GitHub API offer and how do I install it?

Many thanks for your hard work :)

**Reply by Bundy01** [reply]
I found it for the github API.

With the -nW option, I understood that it only searches for subdomains validated by DNS.
Do you have an explanation of how it works without this option? How do the sources perform the search?
